### ISSUE TITLE
Thread-unsafe to use `tf.boolean_mask` in `NormalizeIntensity` in PyGrain dataloader

### ISSUE THREAD

**Issue by innat** [innat/medic-ai#108]
details: https://github.com/tensorflow/tensorflow/issues/123980
